### ISSUE TITLE
[ipmctl] Config file should live in a common location

### ISSUE THREAD

**Issue by sscargal** [intel/ipmctl#35]
Each time ipmctl is executed, it checks for 'ipmctl.conf' in the current working directory and creates it if none exist.  The result is orphaned ipmctl.conf files left all over the filesystem depending on the tool, script, or cwd where ipmctl was executed.

The following example creates a new directory, cd's to it, verifies it's empty, runs an ipmctl command (any options will suffice), then checks the directory contents afterward to find a new file.

```
# mkdir /var/tmp/ipmctl && cd /var/tmp/ipmctl
# ls
# ipmctl show -dimm > /dev/null 2>&1
# ls
ipmctl.conf
```

It would be better to have the ipmctl.conf file live in a common location such as /etc/ipmctl/ipmctl.conf

**Comment by juston-li** [comment]
If ipmctl is installed via an RPM or make install, there will be a conf file installed in /etc/ipmctl.conf and no local conf will be created.

The intent was that if someone compiled and is executing the binary directly without installing, we don't want to touch a system directory like /etc. It's also nice for debug that the conf file is in the same directory as the binary.

From your example, it seems like ipmctl was added to the PATH but not installed?

**Comment by sscargal** [comment]
> From your example, it seems like ipmctl was added to the PATH but not installed?

Correct.  I have a nightly cron job that pulls the ipmctl github repo, builds, and installs it to /usr/local.   

```
# which ipmctl
/usr/local/bin/ipmctl
```

> It's also nice for debug that the conf file is in the same directory as the binary.

Agreed.  However, the `ipmctl.conf` is created in the current working directory Vs the location of the binary (`/usr/local/bin` in my case).  I don't typically work in the same directory each time I run `ipmctl` so I'm left with `ipmctl.conf` files all over my filesystem.  I created a simple addition to the cron job to find and remove unwanted config files.

I have no issues if `ipmctl.conf` was in the basedir(3) location of the ipmctl binary itself for self-compiled use cases (eg: `/usr/local/bin/ipmctl.conf` in my case).  Would that be possible?

**Comment by juston-li** [comment]
> However, the ipmctl.conf is created in the current working directory Vs the location of the binary (/usr/local/bin in my case).

Good point, having it at the location of the binary would be better. Will look into that.

If you remove this from the cmake command
> -DCMAKE_INSTALL_PREFIX=/ ..
 it will install in /usr/local, however we do not look for the conf file in /usr/loca/etc/ipmctl.conf
Will look into this too. Need to pass in the /etc directory rather than hardcoding it.

For now as a workaround, you could move the ipmctl.conf to the /etc directory. Won't have to clean up all the conf files everywhere.

**Comment by ahkok** [comment]
`ipmctl` shouldn't create a conf file - if possible. Ideally it works without any conf file present on the system. If a conf file has to exist, then `ipmctl` should error out if it doesn't exist, and it shouldn't create one in `/etc` unless explicitly directed by the user through some explicit command.

Conf files should never be put in /usr/local/bin. It isn't a `bin` file and it doesn't belong there.

Even if you're self-compiling, you most likely want your `conf` files to go straight into `/etc/` directly.

**Comment by gldiviney** [comment]
Is this still an issue in the current build?

**Comment by rw4s** [comment]
Yes, at least `build.opensuse.org` is complaining about (abbreviated)
```
W: non-etc-or-var-file-marked-as-conffile /usr/share/ipmctl/ipmctl.conf
  A file not in /etc [...] is marked as being a configuration file. Please put
  your conf files in /etc [...].
```
Can you make that a compile-time option -- or simply use that
`CMAKE_INSTALL_SYSCONFDIR` as default (in `src/os/ini/ini.c` and `CMakeLists.txt`)?
And as long as it's (expected to be) only one file, no sub-directory, please!

While you're modifying `CMakeLists.txt`, would it be possible to fix
```
W: incoherent-logrotate-file /etc/logrotate.d/ipmctl.conf
  Your logrotate file should be named /etc/logrotate.d/<package name>.
```
(with `<package name>` being `ipmctl`) or should I open a separate "issue" for that?  Thanks!
